### PR TITLE
Switch from runtime parsing to compile time parsing

### PR DESCRIPTION
**Pull request by pdogr**
Follow-up to #2382

### CODE REVIEW

File: components/calendar/src/any_calendar.rs [hunk 1]
```diff
@@ -39,14 +39,14 @@ use core::fmt;
 /// There are many ways of constructing an AnyCalendar'd date:
 /// ```
 /// use icu::calendar::{AnyCalendar, AnyCalendarKind, DateTime, japanese::Japanese, types::Time};
-/// use icu::locid::Locale;
+/// use icu::locid::{Locale,locale};
```
Comment (sffc) [inline components/calendar/src/any_calendar.rs:42]: Nit, here and elsewhere: I think you don't need to import `Locale` any more. Just `locale`. (I won't comment on every spot)

File: components/datetime/src/any/date.rs [hunk 2]
```diff
@@ -101,14 +101,14 @@ impl DateFormatter {
...
-    /// use icu::locid::Locale;
+    /// use icu::locid::{Locale, locale};
     /// use icu_provider::any::DynamicDataProviderAnyMarkerWrap;
     /// use std::str::FromStr;
     ///
     /// let provider = icu_testdata::get_provider();
     ///
     /// let length = length::Date::Medium;
-    /// let locale = Locale::from_str("en-u-ca-gregory").unwrap();
+    /// let locale: Locale = locale!("en-u-ca-gregory");
```
Comment (sffc) [inline components/datetime/src/any/date.rs:111]: Nit, here and elsewhere: you don't need `: Locale` when you are using `locale!`

**Review by sffc — COMMENTED** [review]
Nice! Two suggestions that apply across all the files

File: components/calendar/src/any_calendar.rs (hunk 1 above)
Comment (pdogr) [reply components/calendar/src/any_calendar.rs]: Done.

File: components/datetime/src/any/date.rs (hunk 2 above)
Comment (pdogr) [reply components/datetime/src/any/date.rs]: Done.

**Review by sffc — APPROVED** [review]

**Pull request merged**